packages/strapi-plugin-upload/services/Upload.js

```diff
@@ -18,7 +18,8 @@ const { bytesToKbytes } = require('../utils/file');
 
 const randomSuffix = () => crypto.randomBytes(5).toString('hex');
 const generateFileName = name => {
-  const baseName = filenamify(name, { replacement: '_' }).replace(/\s/g, '_');
+  // Replace characters that cannot be contained in a file name or are unsafe or reserved in a URL
+  const baseName = filenamify(name, { replacement: '_' }).replace(/[\s#%{}^~[\]`;@=&]/g, '_');
```

Comment:
Maybe we can simply use the slugify utility we have in strapi-utils directly ? would be simpler and cleaner.

Reply:
Sure, that works, although the Upload service tests then fail, as they expect the uppercase characters to be preserved. Would you like me to change these tests or rather preserve the uppercase characters?

Reply:
I would export nameToSlug with an options param so you can customize the behavior to preserve uppercases